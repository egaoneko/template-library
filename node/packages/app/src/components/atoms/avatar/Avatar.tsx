import React, {FC} from 'react';
import styled from 'styled-components/native';

const DEFAULT_IMAGE =
  'https://static.productionready.io/images/smiley-cyrus.jpg';

interface PropsType {
  uri?: string;
  size: number;
}

const Avatar: FC<PropsType> = props => {
  const {uri = DEFAULT_IMAGE, size} = props;
  return <StyledImage source={{uri}} size={size} />;
};

export default Avatar;

const StyledImage = styled.Image<{size: number}>`
  width: ${({size}) => size}px;
  height: ${({size}) => size}px;
  border-radius: ${({size}) => size / 2}px;
`;
