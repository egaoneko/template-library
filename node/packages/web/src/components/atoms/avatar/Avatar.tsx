/* eslint-disable  @next/next/no-img-element */

import React, { FC } from 'react';

interface PropsType {
  url?: string;
  size?: 'small' | 'middle' | 'large';
}

const Avatar: FC<PropsType> = props => {
  let size: number;

  switch (props.size) {
    case 'middle':
      size = 8;
      break;
    case 'large':
      size = 48;
      break;
    case 'small':
    default:
      size = 6;
      break;
  }

  return (
    <div className="flex flex-row justify-center">
      <div className={`relative flex justify-center items-center rounded-full w-${size} h-${size}`}>
        <img className="rounded-full" src={props.url ?? DEFAULT_IMAGE} alt="avatar" />
      </div>
    </div>
  );
};

export default Avatar;

const DEFAULT_IMAGE = 'https://static.productionready.io/images/smiley-cyrus.jpg';
