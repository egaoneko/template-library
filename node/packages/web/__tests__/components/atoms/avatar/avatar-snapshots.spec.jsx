import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../../../../src/components/atoms/avatar/Avatar';

describe('Avatar Snapshots', () => {
  it('renders avatar with default options', () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders avatar with small size', () => {
    const tree = renderer.create(<Avatar size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders avatar with middle size', () => {
    const tree = renderer.create(<Avatar size="middle" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders avatar with large size', () => {
    const tree = renderer.create(<Avatar size="large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders avatar with large size', () => {
    const tree = renderer.create(<Avatar size="large" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders avatar with url', () => {
    const tree = renderer.create(<Avatar url="http://example.com/sample.png" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders avatar with url and size', () => {
    const tree = renderer.create(<Avatar url="http://example.com/sample.png" size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
