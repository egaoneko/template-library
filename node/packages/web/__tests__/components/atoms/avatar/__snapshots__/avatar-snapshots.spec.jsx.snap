// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshots renders avatar with default options 1`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-8 h-8"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;

exports[`Avatar Snapshots renders avatar with large size 1`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-32 h-32"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;

exports[`Avatar Snapshots renders avatar with large size 2`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-32 h-32"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;

exports[`Avatar Snapshots renders avatar with middle size 1`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-8 h-8"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;

exports[`Avatar Snapshots renders avatar with small size 1`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-5 h-5"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.productionready.io%2Fimages%2Fsmiley-cyrus.jpg&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;

exports[`Avatar Snapshots renders avatar with url 1`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-8 h-8"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=3840&q=75"
          srcSet="/_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;

exports[`Avatar Snapshots renders avatar with url and size 1`] = `
<div
  className="Avatar__Wrapper-sc-m9feoe-0 ieHczv"
>
  <div
    className="Avatar__Container-sc-m9feoe-1 fQHyzl w-5 h-5"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="avatar"
        className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="avatar"
          className="Avatar__Img-sc-m9feoe-2 kuaPdK rounded-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=3840&q=75"
          srcSet="/_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=640&q=75 640w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=750&q=75 750w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=828&q=75 828w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2Fexample.com%2Fsample.png&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;
